### PR TITLE
Added a page for downloading mobile application

### PR DESCRIPTION
**Pull request by M4yankChoudhary**
Added  a download page pointing to F-Droid and Play Store.

### CODE REVIEW

File: ihatemoney/templates/layout.html
Comment (Glandos) [inline ihatemoney/templates/layout.html:148]:
```suggestion
            <a data-toggle="tooltip" data-placement="top" title="{{ _('Mobile Application') }}" href="{{url_for('main.downloadMobileApp')}}">
```
Remove attributes needed when opening a new tab. Now, this is an internal link.

File: ihatemoney/static/css/download_mobile_app.css
Comment (Glandos) [inline ihatemoney/static/css/download_mobile_app.css:12]:
```suggestion
    background-color: linear-gradient(150deg, #abe128 0%, #43ca61 100%);
    font-family: 'Comfortaa', arial, serif;
```
Use linear-gradient from home page to be consistent.

File: ihatemoney/static/css/download_mobile_app.css
Comment (Glandos) [inline ihatemoney/static/css/download_mobile_app.css:17]:
```suggestion
```
Use bootstrap class for colors (see HTML)

File: ihatemoney/templates/download_mobile_app.html
Comment (Glandos) [inline ihatemoney/templates/download_mobile_app.html:22]:
```suggestion
<main class="row py-5 justify-content-center text-center">

    <div class=" col-10 text-center bg-dark text-light">
        <h1 class="my-2">{{_("Download Mobile Application")}}</h1>
    </div>
    <h3 class="col-12 my-2">{{_("Get it on")}}</h1>
    <div class="col-12">
        <a target="_blank" rel="noopener" href="https://play.google.com/store/apps/details?id=net.eneiluj.moneybuster&hl=en_US" class="text-decoration-none">
            <img src="{{ url_for("static", filename='images/google-play.png') }}" class="get-it-from"/>
        </a>
    
        <a target="_blank" rel="noopener" href="https://f-droid.org/en/packages/net.eneiluj.moneybuster/" class="text-decoration-none">
            <img src="{{ url_for("static", filename='images/f-droid.svg') }}" class="get-it-from" />
        </a>
    </div>

</main>
```
There are a lot of things here, but mainly:
- Use bootstrap grid system
- Use bootstrap colors and text decoration
- Use bootstrap spacing
- Internationalization

I personally change from `h1` to `h3` to have a smaller `Get it on`, but this is opened to discussion.

**Review by Glandos — CHANGES_REQUESTED** [review]
Very nice! I introduced some changes that I found to be better before merging.
Can you review them and if possible, apply them?

**Comment by M4yankChoudhary** [comment]
Sure, I will do it.

**Review by M4yankChoudhary — COMMENTED** [review]
Resolved

**Review by M4yankChoudhary — COMMENTED** [review]
Added Bootsrap class for color.

**Comment by M4yankChoudhary** [comment]
Added apple-store option for downloading the app.

**Comment by Glandos** [comment]
I made a small modification in the CSS, because I provided the wrong patch :)

I also tried to format web.py, without success.

Is there any reason to include both PNG and SVG for Apple, or is it a simple mistake?

**Comment by M4yankChoudhary** [comment]
> I made a small modification in the CSS, because I provided the wrong patch :)
> 
> I also tried to format web.py, without success.
> 
> Is there any reason to include both PNG and SVG for Apple, or is it a simple mistake?

sorry i forgot to delete the png file. I will do it now.

File: ihatemoney/web.py
Comment (indatwood) [inline ihatemoney/web.py:274]: Hi, I believe you don't need to have the check for the "GET" method here AND on the route decorator. You could remove this from inside the method :-)

File: ihatemoney/web.py
Comment (indatwood) [inline ihatemoney/web.py:273]: The convention we're using elsewhere on the project is to have `camel_case` for the method names. So this should be `download_mobile_app`. In general in python this is what's been used :-)

File: ihatemoney/web.py
Comment (indatwood) [inline ihatemoney/web.py:272]: Should this URL be easier to remember? What about something like just `/mobile`?

File: ihatemoney/web.py
Comment (M4yankChoudhary) [reply ihatemoney/web.py]: Yes you are right! I don't need to check for "GET".

File: ihatemoney/web.py
Comment (M4yankChoudhary) [reply ihatemoney/web.py:272]: Yes done.

File: ihatemoney/static/css/download_mobile_app.css
Comment (indatwood) [inline ihatemoney/static/css/download_mobile_app.css:17]: Hey, it seems you comited some unwanted changes, like these ^

File: ihatemoney/templates/download_mobile_app.html
Comment (indatwood) [inline ihatemoney/templates/download_mobile_app.html:16]: Ditto.

File: ihatemoney/templates/download_mobile_app.html
Comment (indatwood) [inline ihatemoney/templates/download_mobile_app.html:30]: And here again

File: ihatemoney/web.py
Comment (indatwood) [reply ihatemoney/web.py:272]: I don't see the changes in what you pushed. Can you check it? Thanks!

File: ihatemoney/templates/download_mobile_app.html
Comment (M4yankChoudhary) [reply ihatemoney/templates/download_mobile_app.html]: Sorry for my mistakes.

**Comment by M4yankChoudhary** [comment]
I tried to delete the png file but it's not deleting. I thnk only the maintainer can delete this file. 
Screenshot_20201109_152550

File: ihatemoney/templates/layout.html
Comment (Glandos) [inline ihatemoney/templates/layout.html:148]:
```suggestion
            <a data-toggle="tooltip" data-placement="top" title="{{ _('Mobile Application') }}" href="{{url_for('main.mobile')}}">
```

Since the route was rewritten, tests are failing because of that.

**Comment by Glandos** [comment]
@M4yankChoudhary if you use the GitHub UI, you have to go to https://github.com/M4yankChoudhary/ihatemoney/blob/issue597/ihatemoney/static/images/app-store.png and click on the trash icon at the top right.
It seems that I can do it myself, since you provided write access to your branch, but I'll let you do it if you prefer :)

**Comment by indatwood** [comment]
Thanks for the update :-) It seems almost all good for me, once comments by @Glandos will be addressed, thanks!

File: ihatemoney/templates/layout.html
Comment (M4yankChoudhary) [reply ihatemoney/templates/layout.html]: I deleted the png file and resolved conflicts beteween method route.

**Review by Glandos — APPROVED** [review]

**Comment by Glandos** [comment]
So now… GitHub doesn't want to merge because of conflicts. And it seems that some commits are weird branches.

Anyway, you can squash all commits by doing `git reset --soft b89bd1` (the parent commit of your first) in your branch, then commit with a new message, and finally force-push.

Before committing, can you try to execute `make flake8 black` to be sure formatting is OK?

File: ihatemoney/templates/download_mobile_app.html
Comment (Hustenbonbon) [inline ihatemoney/templates/download_mobile_app.html:21]: This has nothing to do with the Android moneybuster app or iHateMoney, it's just a game.

**Comment by M4yankChoudhary** [comment]
> Before committing, can you try to execute `make flake8 black` to be sure formatting is OK?


I tried to run the `make flake8 black` command after installing the flake8 using pip but it says:

make: *** No rule to make target 'flake8'.  Stop.

File: ihatemoney/templates/download_mobile_app.html
Comment (Glandos) [reply ihatemoney/templates/download_mobile_app.html:21]: I feel so noob. Yes, this is not related to our project. So thanks for pointing this and making me ashamed of my own review skills ;)

Anyway, thanks to #697 there is an application for iOS that can be integrated right now:
```suggestion
        <a target="__blank" rel="noopener" href="https://apps.apple.com/us/app/payforme/id1500428306" class="text-decoration-none">
```

**Comment by Glandos** [comment]
I made another mistake: it was `make black isort`. I merged your request like this and format it in #699 so everything is fine.

Thanks for your contribution, it is really appreciated.

**Pull request merged**
